dashboard/app/controllers/application_controller.rb

```diff
@@ -325,6 +328,24 @@ def clear_sign_up_session_vars
     end
   end
 
+  # Check that the user is compliant with the Child Account Policy. If they
+  # are not compliant, then we need to send them to the lockout page.
+  def assert_child_account_policy
+    # Check that the child account policy is currently enabled.
+    return unless CPA.cpa_experience(request)
+
+    # Anonymous users are affected by our Child Account Policy
+    return unless current_user
+    # Don't block any user from signing out
+    return if request.path == '/users/sign_out'
+    # Don't block any user from changing the language
+    return if request.path == '/locale'
+    # Avoid an infinite redirect loop to the lockout page
+    return if request.path == '/lockout'
+    # If a user is not compliant with CAP, restrict them to the lockout page.
+    redirect_to '/lockout' unless current_user.cap_compliant?
```

Comment:
Please don't use hardcoded paths, it's a bad practice.
```suggestion
    return if request.path == destroy_user_session_path
    # Don't block any user from changing the language
    return if request.path == locale_path
    # Avoid an infinite redirect loop to the lockout page
    return if request.path == lockout_path
    # If a user is not compliant with CAP, restrict them to the lockout page.
    redirect_to lockout_path unless current_user.cap_compliant?
```